geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/filter/ArrowFilterOptimizer.scala

```diff
@@ -191,4 +196,12 @@ object ArrowFilterOptimizer extends LazyLogging {
       arrow.getReader(i).asInstanceOf[ArrowDictionaryReader].getEncoded(arrow.getIndex) == value
     }
   }
+
+  case class ArrowListDictionaryEquals(i: Int, value: Int) extends Filter {
```

Comment:
(*Non* blocking question)

Is there a way to check for actual list equality?

Reply:
I think geotools supports that using the filter 'match action' of 'all', but I don't think that's supported through ECQL so effectively there isn't. Maybe there is a filter function, but I'm not aware of one. We could always add one, but again I don't think there is any way to specify a list in ECQL, other than through the default converters (e.g. we could convert the string `"a,b,c"` to `List("a", "b", "c")`)

Reply:
Yeah, I tried to look for use of the Match.ALL.  I believe you are correct!